docs/site/pages/assets/dsl.mdx

```diff
@@ -0,0 +1,126 @@
+---
+title: Writing DSL Components
+---
+
+# Creating TSX Components
+
+In order to take advantage of the auto-completion and validation of TypeScript types, asset libraries can export a component library for content authoring. Creating components isn't much different than writing a React component for the web. The primative elements uses the [react-json-reconciler](https://github.com/intuit/react-json-reconciler) to create the JSON content tree, with utilities to make it quick and painless to create new asset-components.
+
+
+## Creating a Basic Component
+
+The `Asset` component from the `@player-tools/dsl` package is the quickest way to create a new component. The `Asset` component will take all the Asset's properties and convert them to their equivalent JSON representation when serialized.
+
+In the examples below, we'll be creating a TSX component for the `action` asset in our reference set. 
+
+The `action` asset has a `label` slot (which is typically used as a `text` asset), a `value` (for flow transitions), and an `exp` for evaluating expressions.
+It's type resembles something akin to:
+
+```ts
+import { Asset, AssetWrapper, Expression } from '@player-ui/player';
+
+export interface ActionAsset<AnyTextAsset extends Asset = Asset> extends Asset<'action'> {
+  /** The transition value of the action in the state machine */
+  value?: string;
+
+  /** A text-like asset for the action's label */
+  label?: AssetWrapper<AnyTextAsset>;
+
+  /** An optional expression to execute before transitioning */
+  exp?: Expression;
+}
+```
+
+To turn this interface into a usable component, create a new React component that _renders_ an Asset:
+
+```tsx
+import { Asset, AssetPropsWithChildren } from '@player-tools/dsl';
+
+export const Action = (props: AssetPropsWithChildren<ActionAsset>) => {
+  return <Asset type="action" {...props} />;
+}
+```
+
+This would allow users to import the `Action` component, and _render_ it to JSON:
```

Comment:
should we show the DSL and JSON content side by side somewhere, maybe for the Action example below with Label?

Reply:
oh i didnt read the Overview section which has the JSON, so maybe can just link to that section?

Reply:
I think we should definitely inline it. I can add that.